plutus-emulator/src/Wallet/Emulator/Types.hs

```diff
@@ -426,16 +413,18 @@ emLog = view emulatorLog
 -- | Get the blockchain as a list of blocks, starting with the oldest (genesis)
 --   block.
 chainOldestFirst :: Lens' EmulatorState Blockchain
-chainOldestFirst = chainNewestFirst . reversed
+chainOldestFirst = chainState . NC.chainNewestFirst . reversed
 
 type MonadEmulator e m = (MonadError e m, AsAssertionError e, MonadState EmulatorState m)
 
 emptyEmulatorState :: EmulatorState
 emptyEmulatorState = EmulatorState {
-    _chainNewestFirst = [],
-    _txPool = [],
+    _chainState = NC.ChainState {
+        NC._chainNewestFirst = [],
+        NC._txPool           = [],
+        NC._index            = mempty
+    },
```

Comment:
`emptyChainState` seems worth defining. I don't think it should have a monoid instance, though.